src/client/hud.cpp

```diff
@@ -380,16 +381,28 @@ void Hud::drawLuaElements(const v3s16 &camera_offset)
 }
 
 
-void Hud::drawStatbar(v2s32 pos, u16 corner, u16 drawdir, const std::string &texture,
-		s32 count, v2s32 offset, v2s32 size)
+void Hud::drawStatbar(v2s32 pos, u16 corner, u16 drawdir,
+		const std::string &texture, const std::string &bgtexture,
+		s32 count, s32 maxcount, v2s32 offset, v2s32 size)
 {
+	const char separator = '/';
+
 	const video::SColor color(255, 255, 255, 255);
 	const video::SColor colors[] = {color, color, color, color};
 
 	video::ITexture *stat_texture = tsrc->getTexture(texture);
 	if (!stat_texture)
 		return;
 
+	video::ITexture *stat_texture_bg;
+	bool uses_bg_texture = false;
+	if (!bgtexture.empty()) {
+		uses_bg_texture = true;
+		stat_texture_bg = tsrc->getTexture(bgtexture);
+	}
+	if (!stat_texture_bg)
+		uses_bg_texture = false;
```

Comment:
the bool is not necessary here, just use `stat_texture_bg == nullptr` to indicate absence